### PR TITLE
Add k8s ClusterConfig and KubeClient

### PR DESCRIPTION
**Pull request by regadas**
This will be useful for plugins wanting to create resources on remote clusters

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -20,6 +20,8 @@ require (
 	github.com/golang/protobuf v1.4.3
 	github.com/graymeta/stow v0.2.7
 	github.com/hashicorp/golang-lru v0.5.4
+	github.com/lyft/flyteplugins v0.5.26
```
Comment (kumare3) [inline go.mod:23]: this is circular dependency and seems odd. We should not add flyteplugins to flytestdlib. why was this needed?

side note - @katrogan is working on removing lyft/api from the dependency chain.

File: k8s/client.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
+package k8s
```
Comment (kumare3) [inline k8s/client.go:1]: Maybe we should move it to `flyteplugins/pluginmachinery/k8s`

File: go.mod (hunk 1 above)
Comment (regadas) [reply go.mod:23]: Hi @kumare3 ahah yes, I got puzzled as well but haven't yet investigated. Basically, this was just the product of running `go mod tidy`. So there's some dep here that brings it in.

File: k8s/client.go (hunk 2 above)
Comment (regadas) [reply k8s/client.go:1]: That makes sense I was in doubt as well. Should we move the config as well? I think it would make sense actually

**Comment by regadas** [comment]
Closing as I'll be moving this to `flyteplugins`
